File: src/SentenceInReverse.java
import java.util.Scanner;
import java.util.StringTokenizer;

/**
 * Created by USER on 8/20/2017.
 *
 * Print a sentence in Reverse.
 */
public class SentenceInReverse
{
    public void reverseSentence (String sentence)
    {
        //String st = sentence.trim();
        //System.out.println(st);
        String [] array = sentence.split("\\s");
        for (int j=array.length-1; j>=0 ;j--)
        {
            System.out.println(array[j]);
        }

    }

    public static void main(String[] args)
    {
        Scanner s = new Scanner(System.in);
        String d ;
        System.out.println("Enter Sentence ");
        d = s.nextLine();
        SentenceInReverse sr = new SentenceInReverse();
        sr.reverseSentence(d);
    }
}

File: src/algorithms/SentenceInReverse.kt
package algorithms

import java.util.Scanner

/**
 * Created by USER on 8/20/2017.
 *
 * Print a sentence in Reverse.
 */
class SentenceInReverse
{
    fun reverseSentence(sentence: String)
    {
        //String st = sentence.trim();
        //System.out.println(st);
        val array = sentence.split("\\s".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        for (j in array.indices.reversed())
        {
            println(array[j])
        }

    }

    companion object
    {

        @JvmStatic
        fun main(args: Array<String>)
        {
            val s = Scanner(System.`in`)
            val d: String
            println("Enter Sentence ")
            d = s.nextLine()
            val sr = SentenceInReverse()
            sr.reverseSentence(d)
        }
    }
}
